File: frontend/src/types/rag.ts
export interface SourceMetadata {
  page?: number;
  page_label?: string;
  total_pages?: number;
  source?: string;
  creator?: string;
  creationdate?: string;
  moddate?: string;
  producer?: string;
  [key: string]: unknown;
}

export interface QuerySource {
  content: string;
  metadata: SourceMetadata;
}

export interface QueryRequest {
  question: string;
}

export interface QueryResponse {
  question: string;
  answer: string;
  sources: QuerySource[];
}

export interface IngestResponse {
  [key: string]: unknown;
}

File: backend/src/backend/types/rag.py
from typing import Any

from pydantic import BaseModel, Field


class QueryRequest(BaseModel):
    question: str = Field(..., examples=["What is risk management?"])


class DocumentSource(BaseModel):
    content: str
    metadata: dict[str, Any]


class QueryResponse(BaseModel):
    question: str
    answer: str
    sources: list[DocumentSource]


class IngestResponse(BaseModel):
    status: str
    message: str
    chunks_added: int
